### PR TITLE
Update Association REST APIs Swagger Document

### PR DESCRIPTION
**Pull request by tharindu-bandara**

### CODE REVIEW

File: en/docs/develop/restapis/association.yaml [hunk 1]
```diff
@@ -148,49 +153,293 @@ paths:
...
-        -
-          name : SwitchUserReqeust
-          in : body
+        - name: associated-user-id
+          in: path
           required: true
+          type: string
+      summary: Delete given user account from my user associations
+      description: |
+        This API is used to delete the given user account from the associations of the auhtentiated user.
```
Comment (gomathyk) [inline en/docs/develop/restapis/association.yaml]: authenticated

File: en/docs/develop/restapis/association.yaml [hunk 2]
```diff
@@ -148,49 +153,293 @@ paths:
...
+          description: Resource Forbidden
+        500:
+          description: Server Error
           schema:
-             $ref: '#/definitions/Error'
+            $ref: '#/definitions/Error'
+    delete:
+      summary: Delete all my federated user associations
+      description: |
+        This API is used to delete all federated associations of the auhtentiated user.
```
Comment (gomathyk) [inline en/docs/develop/restapis/association.yaml]: all the
authenticated

File: en/docs/develop/restapis/association.yaml [hunk 3]
```diff
@@ -148,49 +153,293 @@ paths:
...
+  /me/federated-associations/{id}:
+    delete:
+      parameters:
+        - name: id
+          in: path
+          required: true
+          type: string
+      summary: Delete my federated association
+      description: |
+        This API is used to delete a federated association of the auhtentiated user.
```
Comment (gomathyk) [inline en/docs/develop/restapis/association.yaml]: authenticated

File: en/docs/develop/restapis/association.yaml [hunk 4]
```diff
@@ -67,17 +72,17 @@ paths:
...
+         <b>Permission required:</b>
+         * /permission/admin/login
       summary: |
         Associate a user to the authenticated user
       parameters:
         -
           name: association
           in: body
-          description:  User details to be associated. userId should be the fully qualified username of the user.
+          description:  User details to be associated. userId should be the fully qalified username of the user.
```
Comment (gomathyk) [inline en/docs/develop/restapis/association.yaml]: qualified

File: en/docs/develop/restapis/association.yaml [hunk 5]
```diff
@@ -67,17 +72,17 @@ paths:
   /me/associations:
     post:
       description: |
-         This API is used to associate a user to the authenticated user. For example, if it is required to associate  my user account to the user account of 'john', this endpoint can be used. userId and the password are required to associate the accounts.
+        This API is used to associate a user to the authenticated user. For example if it is required associate  my user account to the user account of 'john', this endpoint can be used. userId and the password are required to associate the accounts.
```
Comment (gomathyk) [inline en/docs/develop/restapis/association.yaml]: if it is required to associate

File: en/docs/develop/restapis/association.yaml (hunk 1 above)
Comment (tharindu-bandara) [reply en/docs/develop/restapis/association.yaml]: Addressed in 861bcaec1512c2d43091b40273eaf2c2dce11f3e

File: en/docs/develop/restapis/association.yaml (hunk 2 above)
Comment (tharindu-bandara) [reply en/docs/develop/restapis/association.yaml]: Addressed in 861bcaec1512c2d43091b40273eaf2c2dce11f3e

File: en/docs/develop/restapis/association.yaml (hunk 3 above)
Comment (tharindu-bandara) [reply en/docs/develop/restapis/association.yaml]: Addressed in 861bcaec1512c2d43091b40273eaf2c2dce11f3e

File: en/docs/develop/restapis/association.yaml (hunk 4 above)
Comment (tharindu-bandara) [reply en/docs/develop/restapis/association.yaml]: Addressed in 861bcaec1512c2d43091b40273eaf2c2dce11f3e

File: en/docs/develop/restapis/association.yaml (hunk 5 above)
Comment (tharindu-bandara) [reply en/docs/develop/restapis/association.yaml]: Addressed in 861bcaec1512c2d43091b40273eaf2c2dce11f3e

**Pull request merged**